src/main/java/com/igot/cb/community/service/impl/CommunityManagementServiceImpl.java

```diff
@@ -284,4 +290,152 @@ public ApiResponse update(JsonNode communityDetails, String authToken) {
         }
     }
 
+    @Override
+    public ApiResponse joinAndUnjoinCommunity(Map<String, Object> request, String authToken) {
+        log.info("CommunityEngagementService:joinAndUnjoinCommunity:joining");
+        ApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_COMMUNITY_UPDATE);
+        String error = validateJoinPayload(request);
+        if (StringUtils.isNotBlank(error)) {
+            return returnErrorMsg(error, HttpStatus.BAD_REQUEST, response);
+        }
+
+        try {
+            String userId = accessTokenValidator.verifyUserToken(authToken);
```

Comment:
why this method returns "Unauthorized" ?? just return empty or return proper user id value.